cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -1649,6 +1710,32 @@ fn x64_map_regs<RUM: RegUsageMapper>(inst: &mut Inst, mapper: &RUM) {
             src.map_uses(mapper);
             map_def(mapper, dst);
         }
+        Inst::XmmRmRImm {
+            ref mut src,
+            ref mut dst,
+            ref op,
+            ..
+        } => {
+            // In certain cases, instructions of this format can convert an XMM register into a
+            // define (e.g. an equality comparison); this extra logic is necessary to inform the
+            // registry allocator of a different register usage.
+            if let RegMem::Reg { reg } = src {
+                if (*op == SseOpcode::Cmppd || *op == SseOpcode::Cmpps) && *reg == dst.to_reg() {
```

Comment:
Given that this logic is duplicated, let's pull it out into a helper to ensure the two locations stay in sync.

Reply:
This is being done in #2071; we need to figure out a merge ordering for these two PRs.